pkg/providers/common/common.go

```diff
@@ -67,13 +68,20 @@ func ParseSSHAuthKey(key *string) error {
 }
 
 func GenerateSSHAuthKey(username string, writer filewriter.FileWriter) (string, error) {
-	keygenerator, _ := crypto.NewKeyGenerator(writer)
-	sshAuthorizedKeyBytes, err := keygenerator.GenerateSSHKeyPair("", "", privateKeyFileName, publicKeyFileName, username)
-	if err != nil || sshAuthorizedKeyBytes == nil {
+	privateKeyPath, sshAuthorizedKeyBytes, err := crypto.NewSshKeyPairUsingFileWriter(writer, privateKeyFileName, publicKeyFileName)
```

Comment:
You may need to pull the latest changes we made here to make this a persistent file. 

https://github.com/aws/eks-anywhere/pull/1395

Reply:
merged latest changes with newest commit